lib_eio/net.ml

```diff
@@ -267,3 +269,52 @@ let with_tcp_connect ?(timeout=Time.Timeout.none) ~host ~service t f =
   | exception (Exn.Io _ as ex) ->
     let bt = Printexc.get_raw_backtrace () in
     Exn.reraise_with_context ex bt "connecting to %S:%s" host service
+
+let accept_fork_domain ~on_error (domain_mgr, domains) (t : #listening_socket) connection_handler () =
+  Switch.run @@ fun sw ->
+  let flow, addr = accept ~sw t in
+  Fun.protect
+    ~finally:(fun () ->
+      Flow.close flow;
```

Comment:
Hmm... Is this `close` necessary?  The documentation [seems to suggests](https://github.com/ocaml-multicore/eio#networking) that a switch automatically closes attached resources.

Reply:
Good catch. Indeed, it doesn't need to be.